invenio_rdm_records/services/components/pids.py

```diff
@@ -244,10 +243,27 @@ def edit(self, identity, draft=None, record=None):
         # FIXME: this should be moved to marshmallow level
         # validation happens before to be able to fail before reserving pids
         # it means a double looping but validation should eventually be moved
-        # to marshmellow see invenio-rdm-records/issues/796
-        errors = {}
+        # to marshmallow see invenio-rdm-records/issues/796
+        errors = []
         self._validate_pids_schemes(record_pids)
         self._validate_pids(record_pids, record, errors)
         if errors:
             raise ValidationError(message=errors)
         draft.pids = record_pids
+
+    def post_publish(self, identity, record=None):
+        """Post publish handler."""
+        service = current_rdm_records.records_service
+        record_pids = record.get('pids', {})
+        for scheme, pid_dict in record_pids.items():
+            provider_name = pid_dict.get("provider")
+            identifier_value = pid_dict.get("identifier")
```

Comment:
At this point, *after published*, pids **must** have the `provider` and `identifier` attributes. So we can access with `[]` so it fails earlier (not below with a `PIDDoesNotExists` later on)